step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/TestSetHandler.java

```diff
@@ -33,12 +35,26 @@
 import java.util.OptionalInt;
 import java.util.function.Consumer;
 
+import static step.artefacts.handlers.functions.TokenForcastingExecutionPlugin.getTokenForecastingContext;
+import static step.artefacts.handlers.functions.TokenForcastingExecutionPlugin.pushNewTokenNumberCalculationContext;
+
 public class TestSetHandler extends ArtefactHandler<TestSet, ReportNode> {
 	
 	@Override
 	public void createReportSkeleton_(ReportNode node, TestSet testSet) {	
 		context.getExecutionManager().updateExecutionType("TestSet");
-		runParallel(node, testSet, false);
+
+		TokenForecastingContext tokenForecastingContext = getTokenForecastingContext(context);
+		MaxAndMultiplyingTokenForecastingContext newTokenForecastingContext = new MaxAndMultiplyingTokenForecastingContext(tokenForecastingContext, testSet.getThreads().getOrDefault(1));

```

Comment:
Here we should reuse the same logic as the one used during execution

Reply:
Done